library(QDNAseq)
library(DNAcopy)
# bins <- getBinAnnotations(binSize=100, genome="hg19")
# saveRDS(bins,file = "bin_annotations_100kbp.rds")
bins <- readRDS("bin_annotations_100kbp.rds")
# bins <- readRDS("bin_annotations_10kbp.rds")

files <- list.files(path="/home/garner1/Work/dataset/WGS", pattern = "\\.bam$", full.names=T, recursive=FALSE)
# files <- list.files(path="/home/garner1/Work/dataset/CNV", pattern="bam$", full.names=T, recursive=FALSE)

for (x in files){
  readCounts <- binReadCounts(bins,bamfiles=x)  
  readCountsFiltered <- applyFilters(readCounts,residual=TRUE, blacklist=TRUE)
  readCountsFiltered <- estimateCorrection(readCountsFiltered)
  copyNumbers <- correctBins(readCountsFiltered)
  copyNumbersNormalized <- normalizeBins(copyNumbers)
  copyNumbersSmooth <- smoothOutlierBins(copyNumbersNormalized)
  copyNumbersSegmented <- segmentBins(copyNumbersSmooth, transformFun="sqrt",smoothBy = 1L)
  copyNumbersSegmented <- normalizeSegmentedBins(copyNumbersSegmented)
  copyNumbersCalled <- callBins(copyNumbersSegmented,method = "cutoff")
  filename<- paste(x, ".png", sep="")
  png(filename=filename)
  plot(copyNumbersCalled)  
  dev.off()
  filename <- paste(x,".bed", sep="")
  exportBins(copyNumbersSegmented, filename, format = "bed")  
  exportBins(copyNumbersCalled, format="vcf")
  filename <- paste(x,".bed", sep="")
  exportBins(copyNumbersCalled, filename, type = "segments", format="tsv")
}

plot(readCounts, logTransform=FALSE, ylim=c(-10, 200))
highlightFilters(readCounts, logTransform=FALSE,residual=TRUE, blacklist=TRUE)
readCountsFiltered <- applyFilters(readCounts,residual=TRUE, blacklist=TRUE)
readCountsFiltered <- estimateCorrection(readCountsFiltered)
copyNumbers <- correctBins(readCountsFiltered)
copyNumbersNormalized <- normalizeBins(copyNumbers)
copyNumbersSmooth <- smoothOutlierBins(copyNumbersNormalized)
plot(copyNumbersSmooth,ylim=c(-2, 2))
copyNumbersSegmented <- segmentBins(copyNumbersSmooth, transformFun="sqrt",smoothBy = 1L)
copyNumbersSegmented <- normalizeSegmentedBins(copyNumbersSegmented)
plot(copyNumbersSegmented,ylim=c(-2,2))
######################################################
readCounts <- binReadCounts(bins,bamfiles='/home/garner1/Work/dataset/WGS/MN1/MN1.bam')  
readCountsFiltered <- applyFilters(readCounts,residual=TRUE, blacklist=TRUE)
readCountsFiltered <- estimateCorrection(readCountsFiltered)
copyNumbers <- correctBins(readCountsFiltered)
copyNumbersNormalized <- normalizeBins(copyNumbers)
copyNumbersSmooth <- smoothOutlierBins(copyNumbersNormalized)
copyNumbersSegmented <- segmentBins(copyNumbersSmooth, transformFun="sqrt",smoothBy = 1L)
copyNumbersSegmented <- normalizeSegmentedBins(copyNumbersSegmented)
# filename<- paste(x, ".png", sep="")
# png(filename=filename)
plot(copyNumbersSegmented,main="MN1")  
# dev.off()
##########################################################
readCounts <- binReadCounts(bins,bamfiles="/home/garner1/Work/dataset/reduced_sequencing/cellLines/NlaIII/bamfiles/XZ64_CATCACGC.deduplicated.bam")  
readCountsFiltered <- applyFilters(readCounts,residual=TRUE, blacklist=TRUE)
readCountsFiltered <- estimateCorrection(readCountsFiltered)
copyNumbers <- correctBins(readCountsFiltered)
copyNumbersNormalized <- normalizeBins(copyNumbers)
copyNumbersSmooth <- smoothOutlierBins(copyNumbersNormalized)
copyNumbersSegmented <- segmentBins(copyNumbersSmooth, transformFun="sqrt",smoothBy = 1L)
copyNumbersSegmented <- normalizeSegmentedBins(copyNumbersSegmented)
copyNumbersCalled <- callBins(copyNumbersSegmented)
filename<- paste(x, ".png", sep="")
png(filename=filename)
plot(copyNumbersCalled)  
dev.off()
filename <- paste(x,".bed", sep="")
exportBins(copyNumbersSegmented, filename, format = "bed")  
exportBins(copyNumbersCalled, format="vcf")
